receiver/mysqlreceiver/scraper.go

```diff
@@ -99,10 +99,83 @@ func (m *mySQLScraper) scrape(context.Context) (pmetric.Metrics, error) {
+		return
+	}
+
+	for i := 0; i < len(tableIoWaitsStats); i++ {
+		s := tableIoWaitsStats[i]
+		// counts
+		m.mb.RecordMysqlTableIoWaitCountDataPoint(now, s.countDelete, metadata.AttributeIoWaitsOperationsDelete, s.name, s.schema)
+		m.mb.RecordMysqlTableIoWaitCountDataPoint(now, s.countFetch, metadata.AttributeIoWaitsOperationsFetch, s.name, s.schema)
+		m.mb.RecordMysqlTableIoWaitCountDataPoint(now, s.countInsert, metadata.AttributeIoWaitsOperationsInsert, s.name, s.schema)
+		m.mb.RecordMysqlTableIoWaitCountDataPoint(now, s.countUpdate, metadata.AttributeIoWaitsOperationsUpdate, s.name, s.schema)
+
+		// times
+		m.mb.RecordMysqlTableIoWaitTimeDataPoint(
+			now, s.timeDelete/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsDelete, s.name, s.schema,
+		)
+		m.mb.RecordMysqlTableIoWaitTimeDataPoint(
+			now, s.timeFetch/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsFetch, s.name, s.schema,
+		)
+		m.mb.RecordMysqlTableIoWaitTimeDataPoint(
+			now, s.timeInsert/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsInsert, s.name, s.schema,
+		)
+		m.mb.RecordMysqlTableIoWaitTimeDataPoint(
+			now, s.timeUpdate/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsUpdate, s.name, s.schema,
+		)
+	}
+}
+
+func (m *mySQLScraper) scrapeIndexIoWaitsStats(now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
+	indexIoWaitsStats, err := m.sqlclient.getIndexIoWaitsStats()
+	if err != nil {
+		m.logger.Error("Failed to fetch index io_waits stats", zap.Error(err))
+		errs.AddPartial(8, err)
+		return
+	}
+
+	for i := 0; i < len(indexIoWaitsStats); i++ {
+		s := indexIoWaitsStats[i]
+		// counts
+		m.mb.RecordMysqlIndexIoWaitCountDataPoint(now, s.countDelete, metadata.AttributeIoWaitsOperationsDelete, s.name, s.schema, s.index)
+		m.mb.RecordMysqlIndexIoWaitCountDataPoint(now, s.countFetch, metadata.AttributeIoWaitsOperationsFetch, s.name, s.schema, s.index)
+		m.mb.RecordMysqlIndexIoWaitCountDataPoint(now, s.countInsert, metadata.AttributeIoWaitsOperationsInsert, s.name, s.schema, s.index)
+		m.mb.RecordMysqlIndexIoWaitCountDataPoint(now, s.countUpdate, metadata.AttributeIoWaitsOperationsUpdate, s.name, s.schema, s.index)
+
+		// times
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeDelete/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsDelete, s.name, s.schema, s.index,
+		)
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeFetch/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsFetch, s.name, s.schema, s.index,
+		)
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeInsert/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsInsert, s.name, s.schema, s.index,
+		)
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeUpdate/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsUpdate, s.name, s.schema, s.index,
+		)
+	}
+}
+
+func (m *mySQLScraper) scrapeGlobalStats(now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
```

Comment:
For the sake of a cleaner diff, will you please move this function definition above the table and index functions?

Reply:
I did this. I should have noticed it before, sorry.